app/src/main/java/ml/docilealligator/infinityforreddit/markdown/SuperscriptPlugin.java

```diff
@@ -0,0 +1,158 @@
+
+public class SuperscriptPlugin extends AbstractMarkwonPlugin {
+    private final SuperscriptOpeningStorage superscriptOpeningBracketStorage;
+    private final List<SuperscriptOpening> superscriptOpeningList;
+
+    SuperscriptPlugin() {
+        this.superscriptOpeningBracketStorage = new SuperscriptOpeningStorage();
+        this.superscriptOpeningList = new ArrayList<>();
+    }
+
+    @NonNull
+    public static SuperscriptPlugin create() {
+        return new SuperscriptPlugin();
+    }
+
+    private static char peek(int index, CharSequence input) {
+        return index >= 0 && index < input.length() ? input.charAt(index) : '\0';
+    }
+
+    private static SpannableBuilder.Span matchSpanAtPosition(List<SpannableBuilder.Span> list, int value, Class spanClass) {
+        for (var span : list)
+            if (span.what.getClass() == spanClass && span.start <= value && value <= span.end)
+                return span;
+        return null;
+    }
+
+    private static SpannableBuilder.Span matchNonTextSpanAtBoundary(List<SpannableBuilder.Span> list, int value) {
+        for (var span : list)
+            if (span.what.getClass() != TextViewSpan.class && (span.end == value || span.start == value))
+                return span;
+        return null;
+    }
+
+    @Override
+    public void configureSpansFactory(@NonNull MarkwonSpansFactory.Builder builder) {
+        builder.setFactory(Superscript.class, (config, renderProps) -> new SuperscriptSpan());
+    }
+
+    @Override
+    public void configureVisitor(@NonNull MarkwonVisitor.Builder builder) {
+        builder.on(Superscript.class, new MarkwonVisitor.NodeVisitor<>() {
+            int depth = 0;
+
+            @Override
+            public void visit(@NonNull MarkwonVisitor visitor, @NonNull Superscript superscript) {
+                int start = visitor.length();
+
+                if (!superscript.isBracketed()) {
+                    superscriptOpeningList.add(new SuperscriptOpening(superscript, start));
+                    return;
+                }
+
+                depth++;
+                visitor.visitChildren(superscript);
+                depth--;
+                if (depth == 0) {
+                    int end = visitor.builder().length();
+                    var spannableStringBuilder = visitor.builder().spannableStringBuilder();
+                    var codeSpans = spannableStringBuilder.getSpans(start, end, CodeSpan.class);
+                    if (codeSpans.length > 0) {
+                        for (CodeSpan span : codeSpans) {
+                            if (spannableStringBuilder.getSpanEnd(span) <= end) {
+                                visitor.builder().setSpan(new SuperscriptSpan(), start, spannableStringBuilder.getSpanStart(span));
+                                visitor.builder().setSpan(new SuperscriptSpan(), spannableStringBuilder.getSpanEnd(span), end);
+                            }
+                        }
```

Comment:
I think this doesn't work as you intended with multiple code spans, e.g. ``a^(b `c` d `e` f)``

Reply:
Thanks!